src/viam/sdk/module/service.cpp

```diff
@@ -44,14 +44,13 @@
 namespace viam {
 namespace sdk {
 
-namespace {
```

Comment:
I'm not sure why this was in an unnamed namespace; happy to keep it that way if desired, but I moved it to be a private method on `ModuleService`.

Reply:
I'm sure there was a reason but I can't recall what it is. Having it as a private method seems much better.

Reply:
Well, technically, it doesn't need any access to private state of the `ModuleService`, so it doesn't strictly need to be a member. In general, if something doesn't need to be a member of a class, it shouldn't be. I think this is basically an "implementation" helper needed only in this file to support defining `[Add|Reconfigure]Resource`. By making it a private member of `ModuleService`, it now has linkage/visibility even though it doesn't need it.

Reply:
> Well, technically, it doesn't need any access to private state of the ModuleService, so it doesn't strictly need to be a member.

I think I was looking at `get_dependencies` and noticed it took a `ModuleService_* m`; that _looks_ a lot like a method on `ModuleService_` and not necessarily a helper function. I do see what you're saying about it not accessing private state, though. Happy to change back if you feel strongly.

Reply:
No, I don't feel strongly, its fine as a private member.